### PR TITLE
Arm neon port

### PR DESCRIPTION
**Pull request by phlash**
Hi ExtIO_sddc maintainers, I had an interest in running this on Linux / ARM (RaspberryPi, OrangePi, etc.) as a downsampling head-end, so have made the necessary adjustments to the build and added a Neon-optimised R2IQ implementation.

There are also a few changes to the (out-of-date / incompatible) `unittest/core_test.cpp` file, as I based my work on v1.2.0 tagged build (I found that current version fails to pass unit tests on x86, crashing with a pointer problem, hence starting with last stable). Not sure what you would like to do here, please let me know?

### CODE REVIEW

**Comment by ik1xpv** [comment]
Hi Phlash, thank you. I appreciate your input.  My programming skills are limited and I have no experience with Linux and Arm. Howard's, Hayati's, Franco's ideas and coding made it possible to get here.
My activity is reduced in Summer, when I am a full time grandfather :-)

**Comment by phlash** [comment]
Congratulations on grandfather-hood, always a priority :smile: 

I'm happy to wait for @howard0su, @hayguen or @fventuri to respond, while I continue into real hardware and testing!

File: unittest/core_test.cpp [hunk 1]
```diff
@@ -118,12 +146,14 @@ TEST_CASE(CoreFixture, R2IQTest)
...
         totalsize = 0;
         radio->Start(decimate); // full bandwidth
-        std::this_thread::sleep_for(0.5s);
+        std::this_thread::sleep_for(1s);
         radio->Stop();
 
         REQUIRE_TRUE(count > 0);
         REQUIRE_TRUE(totalsize > 0);
         REQUIRE_TRUE(totalsize / count == transferSamples / 2);
+    printf("decimate=%d nxfers=%ld mindelay=%ld count=%u totalsize=%" PRIu64 "\n",
```
Comment (howard0su) [inline unittest/core_test.cpp:155]: space

File: wisdom-arm [hunk 2]
```diff
@@ -0,0 +1,61 @@
+(fftw-3.3.8 fftwf_wisdom #x4a633eef #xb5a95564 #x91014bdd #x9c85ce5f
```
Comment (howard0su) [inline wisdom-arm:1]: do we really need the wisdom files?

**Review by howard0su — CHANGES_REQUESTED** [review]
Looks good to me except wisdom files. Not sure if we want to commit into repo. My understanding is that the files are changed for different hardware.

File: unittest/core_test.cpp (hunk 1 above)
Comment (phlash) [reply unittest/core_test.cpp:155]: Gah, VS code not doing quite the right thing for me, will fix.

**Comment by phlash** [comment]
> Looks good to me except wisdom files. Not sure if we want to commit into repo. My understanding is that the files are changed for different hardware.

Thanks, this is true, I was simply saving them as it takes several minutes to generate on my wee ARM board. I'll pop a pattern in .gitignore and remove these from the PR.

**Comment by howard0su** [comment]
Can you resolve the conflict of unittest? then we can merge.

**Comment by phlash** [comment]
> Can you resolve the conflict of unittest? then we can merge.

This should be good to merge now, although I am still unable to run `unittest` due to the segfault somewhere in `R2IQ_impl` I mentioned at the start. Would you like me to raise a new issue for that?

**Pull request merged**